app/models/harvest.rb

```diff
@@ -118,4 +138,13 @@ def crop_name_to_human
   def default_photo
     photos.first || crop.default_photo
   end
+
+  def update_predictions
+    return if planting.blank?
+    # Set the actual harvest on the planting
+    planting.update(harvested_at: Harvest.where(planting: self).order(:harvested_at).first.harvested_at)
+
+    differences = Planting.where(crop: crop, finished: false).collect(&:time_to_first_harvest)
+    differences.compact.sum / differences.compact.size unless differences.compact.empty?
```

Comment:
- The three calls to `compact` make this IMHO slightly less readable - I'd either put one call on the line above, or add a new line `differences = differences.compact`.
 - It looks like we're loading in every unfinished planting of a given crop (why only unfinished plantings?) and calling a method on it, then calculating the mean in Ruby. This is probably pretty inefficient. Can we calculate the mean using SQL aggregations?
 - These two lines are AFAICT equivalent to `PlantingPrediction.mean_days_to_first_harvest(Planting.where(crop: crop, finished: false)`, and code duplication, however slight, makes me twitchy.
 - This method doesn't seem to actually update the predictions? We're returning the new mean-time-to-harvest, but AFAICT the return value of this method is always thrown away.